### PR TITLE
Upgrade packages of requirements and use poetry extra to manage multi backend packages

### PR DESCRIPTION
**Pull request by lust4life**
After changing:

```
click: 6.7 => 8.1.3
clickhouse-driver: 0.2.3 => 0.2.4
clickhouse-sqlalchemy: 0.1.6 => 0.2.1
colorlog: 4.0.2 => 6.6.0
coverage: 6.3.2 => 6.4.3
psycopg2-binary: 2.8.6 => 2.9.3
pymongo: 3.10.1 => 4.2.0
pyspark: 3.1.3 => 3.3.0
regex: 2022.6.2 => 2022.7.25
SQLAlchemy: 1.3.23 => 1.4.40
sqlfluff: 1.2.1 => not changed
xlrd: 1.2.0 => not changed
```

code changes are mainly in the `rdb.py` file, caused by `SQLAlchemy`, as they moved `ResultProxy` to another module, and return a `KeyViews` type instead of `list(str)` when calling its `.keys` method.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -21,8 +21,15 @@ jobs:
...
-    - run: make package-pip
-  
+    - name: Load cached Poetry installation
+      uses: actions/cache@v2
+      with:
+        path: ~/.local  # the path depends on the OS
+        key: poetry-1  # increment to reset cache
+    - name: Install Poetry
+      uses: snok/install-poetry@v1.3.1
+    - run: poetry build
```
Comment (gmlove) [inline .github/workflows/build.yaml:31]: 这里可以实现跟package-pip一样的功能？

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+max-line-length = 88
```
Comment (gmlove) [inline .flake8:2]: 这里可以考虑调成120或150吧，88有点太小了

File: .github/workflows/build.yaml (hunk 1 above)
Comment (lust4life) [reply .github/workflows/build.yaml:31]: I have tried it locally and compared the generated `easy_sql-easy_sql-0.3.0.tar.gz  easy_sql_easy_sql-0.3.0-py3-none-any.whl` files, they are almost identical, as we haven't included any complex build process here.

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[flake8]
+max-line-length = 120
+extend-ignore = E203, E501
```
Comment (gmlove) [inline .flake8:3]: 字符串有很多单引号到双引号的change，这个我觉得可以保持用单引号吧

Comment (lust4life) [reply .flake8:3]: try to use black as the formatter, https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#strings , if we are not going to respect this rule, we could do it with: `--skip-string-normalization` .  Might be a good idea to try to normalize/format the whole project in one shot, then later changes will be easy to review, and no more formatting changes to disturb the focus.

Comment (gmlove) [reply .flake8:3]: That's right. We can do it in a later change. For this pr, can we revert the quotes change. Also, is there a way to format code during file saving?

Comment (lust4life) [reply .flake8:3]: > For this pr, can we revert the quotes change
Sure, will do it later. 

>  Also, is there a way to format code during file saving?
Yep, vscode python extension will let you choose format on save with invoking black, and we can also setup a pre-commit hook or github-checks to make it applied to all the later commits

Comment (lust4life) [reply .flake8:3]: Changed back to single quotes.

**Pull request merged**